apps/console/src/app/providers.tsx

```diff
@@ -5,21 +5,29 @@ import { createClient, createSubscriberClient } from '@/lib/urql'
 import { useSession } from 'next-auth/react'
 import { ThemeProvider } from '@/providers/theme'
 import { usePathname } from 'next/navigation'
+import { useMemo, useState, useEffect } from 'react'
 
 interface ProvidersProps {
   children: any
 }
 
 const Providers = ({ children }: ProvidersProps) => {
-  const { data: session } = useSession()
+  const { data: session, status } = useSession()
   const pathname = usePathname()
-  var client = createClient(session)
+  const [client, setClient] = useState<any>(null)
 
-  // override client for waitlist page
-  // this uses an API token instead of the user's credentials
-  if (pathname.endsWith('waitlist')) {
-    client = createSubscriberClient()
-  }
+  useEffect(() => {
+    if (status === 'authenticated' && !client) {
+      setClient(createClient(session))
+    } else if (status === 'unauthenticated' && pathname.endsWith('waitlist')) {
+      setClient(createSubscriberClient())
+    }
+  }, [session, status, pathname])
+
+  console.log('status', status)
+  if (status === 'loading') return null
+  if (status === 'authenticated' && !client) return null
+  console.log('client', client)
```

Comment:
Do we need this logging or was this for debugging?

```suggestion
  if (status === 'loading') return null
  if (status === 'authenticated' && !client) return null
```

Reply:
removed